react/components/Router.js

```diff
@@ -209,17 +212,21 @@ export class Router extends Component {
     }
   }
 
-  changePage(location) {
-    const path = location.pathname
-    const page = this.pageNameFromPath(path)
+  changePage(location, action) {
+    const {pathname, search, state = {}} = location
+    if (!state.renderRouting && global.__RUNTIME__.customRouting) {
```

Comment:
Shouldn't this be an `or`? `!state.renderRouting || global.__RUNTIME__.customRouting`. Dessa forma se você tiver `Link`s dentro mas tiver habilitado o customRouting, ainda é respeitado

Reply:
Eu acredito que se alguém de alguma forma está navegando sem usar nosso componente `Link`, está navegando por conta própria e nosso roteamento deveria ser ignorado.

Eu coloquei esse `customRouting` apenas para evitar que isso seja uma breaking change. Vai que exista alguma app que se apoia no fato de que independente de como a navegação é feita, o render-runtime lida com ela?

Reply:
Não existe. Mas acho que tem que ter, sim, esse customRouting. A única coisa que to dizendo é que acho que a navegação não tem que ser nossa se _qualquer um dos critérios_ acontecer: ou não foi nosso Link, ou mesmo que tenha sido, alguém declarou o customRouting.

Mas, pensando um pouco a mais, se vc botar um href sem ser Link num SPA do render mas a rota destino é uma page que temos como navegar, não vamos navegar?

Reply:
OK. Se não houver `customRouting=true` esse caso que você falou funciona. Apesar de achar que deveríamos forçar o uso o `Link`.

Então de fato `customRouting` é necessário, não apenas pra evitar breaking change.

Pensei também que seria possível ao invés de ter esse booleano, ter informação de exatamente quais rotas você quer tratar sem interferência do runtime, mas acho que isso pode ser uma evolução futura. Ou não, já que essa feature de customRouting é no mínimo controversa.